### ISSUE TITLE
Do not depend on cobertura plugin

### ISSUE THREAD

**Issue by kwisatz** [uber/phabricator-jenkins-plugin#117]
Even when the "Uberalls" checkbox is not set, the build will fail if the cobertura plugin is not installed in Jenkins.

```bash
[phabricator:plugin-provider] 'cobertura' plugin not installed.
ERROR: Step ‘Post to Phabricator’ aborted due to exception: 
java.lang.NoClassDefFoundError: hudson/plugins/cobertura/CoberturaBuildAction
	at com.uber.jenkins.phabricator.coverage.CoberturaCoverageProvider.getCoverageResult(CoberturaCoverageProvider.java:89)
	at com.uber.jenkins.phabricator.coverage.CoberturaCoverageProvider.hasCoverage(CoberturaCoverageProvider.java:46)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.getCoverageProvider(PhabricatorNotifier.java:216)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.perform(PhabricatorNotifier.java:89)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:723)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:668)
	at hudson.model.Run.execute(Run.java:1763)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: java.lang.ClassNotFoundException: hudson.plugins.cobertura.CoberturaBuildAction
	at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1376)
	at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
	at jenkins.util.AntClassLoader.loadClass(AntClassLoader.java:1079)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
```

**Comment by sectioneight** [comment]
That is definitely not the intended behavior. Thanks for the report, I'll take a look at this one soon.

**Comment by sectioneight** [comment]
Which version of the plugin are you running? The stacktrace above appears out of date. It should be using `InstanceProvider` which safely checks whether the plugin is installed or not before trying to load the class.

**Comment by kwisatz** [comment]
We're using 1.8.3.
There's no upgrade visible in the plugin manager though.

**Comment by sectioneight** [comment]
My bad, found the bug. Was a really dumb one :)

I'll cut 1.8.4 in the next day or so (looking to merge #108 as well).

Thanks!

**Comment by sectioneight** [comment]
This is merged and released in 1.9.0, which I just uploaded to maven. You should see it in the your update manager within the next 12 hours.

**Comment by kwisatz** [comment]
Thanks, plugin updated and running!

**Comment by jannet-faiz** [comment]
I am running with the latest available plugin version 1.9.7 and running into the same issue.
Could you please help me out here?
